app/scripts/services/tile-proxy.js

```diff
@@ -533,43 +528,45 @@ export const tileDataToPixData = (
 
 function text(url, callback) {
   /**
-   * Send a JSON request mark it so that we can tell how many are in flight
+   * Send a text request and mark it so that we can tell how many are in flight
    */
-  requestsInFlight += 1;
-  pubSub.publish('requestSent', url);
-
-  return fetch(url)
-    .then(rep => rep.text())
-    .then((text) => {
-      callback(undefined, text);
-    })
-    .catch((error) => {
-      callback(error, undefined);
-    })
-    .finally(() => {
-      pubSub.publish('requestReceived', url);
-      requestsInFlight -= 1;
-    });
+  fetchEither(url, callback, 'text');
```

Comment:
`return` is missing. Callbacks are old-school but we need them for compatibility but we should also return the promise such that we can progressively update the code base.